//gapis/api/vulkan:go_default_test panics

```
--- FAIL: TestCallReflectedCommand (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x16bbebe]

goroutine 20 [running]:
testing.tRunner.func1(0xc4203f40f0)
	GOROOT/src/testing/testing.go:742 +0x29d
panic(0x24a3a00, 0x3ee0480)
	GOROOT/src/runtime/panic.go:505 +0x229
github.com/google/gapid/gapis/api/vulkan.QueryPoolObjectʳ.SetLastBoundQueue(...)
	bazel-out/k8-fastbuild/genfiles/gapis/api/vulkan/api.go:30017
github.com/google/gapid/gapis/api/vulkan.subDovkCmdWriteTimestamp(0x2a7c7a0, 0xc4203eacc0, 0x2a8d820, 0xc42016ea20, 0xa, 0xc4203eadb0, 0xc4203369a0, 0xc4202d0dc0, 0x0, 0x0, ...)
	bazel-out/k8-fastbuild/genfiles/gapis/api/vulkan/mutate.go:2388 +0x6e
reflect.Value.call(0x25886e0, 0x288abe0, 0x13, 0x282ee6e, 0x4, 0xc42010ae30, 0x9, 0x9, 0xc42033a968, 0x2740ac0, ...)
	GOROOT/src/reflect/value.go:447 +0x969
reflect.Value.Call(0x25886e0, 0x288abe0, 0x13, 0xc42010ae30, 0x9, 0x9, 0xc4200f95d0, 0x9c9929, 0xc420336930)
	GOROOT/src/reflect/value.go:308 +0xa4
github.com/google/gapid/gapis/api/vulkan.CallReflectedCommand(0x2a7c7a0, 0xc4203eacc0, 0x2a8d820, 0xc42016ea20, 0xa, 0xc4203369a0, 0x0, 0x25886e0, 0x288abe0, 0x2740ac0, ...)
	gapis/api/vulkan/externs.go:70 +0x530
github.com/google/gapid/gapis/api/vulkan.TestCallReflectedCommand(0xc4203f40f0)
	gapis/api/vulkan/externs_test.go:40 +0x1af
testing.tRunner(0xc4203f40f0, 0x2887f20)
	GOROOT/src/testing/testing.go:777 +0xd0
created by testing.(*T).Run
	GOROOT/src/testing/testing.go:824 +0x2e0
```

--- a/gapis/api/vulkan/api/query_pool.api
+++ b/gapis/api/vulkan/api/query_pool.api
@@ -132,8 +132,10 @@ cmd VkResult vkGetQueryPoolResults(
 
 sub void dovkCmdBeginQuery(ref!vkCmdBeginQueryArgs args) {
   pool := QueryPools[args.QueryPool]
-  pool.Status[args.Query] = QUERY_STATUS_ACTIVE
-  pool.LastBoundQueue = LastBoundQueue
+  if pool != null {
+    pool.Status[args.Query] = QUERY_STATUS_ACTIVE
+    pool.LastBoundQueue = LastBoundQueue
+  }
 }
 
 @threadSafety("app")
@@ -164,8 +166,10 @@ vkCmdEndQueryArgs {
 
 sub void dovkCmdEndQuery(ref!vkCmdEndQueryArgs args) {
   pool := QueryPools[args.QueryPool]
-  pool.Status[args.Query] = QUERY_STATUS_COMPLETE
-  pool.LastBoundQueue = LastBoundQueue
+  if pool != null {
+    pool.Status[args.Query] = QUERY_STATUS_COMPLETE
+    pool.LastBoundQueue = LastBoundQueue
+  }
 }
 
 @threadSafety("app")
@@ -196,10 +200,12 @@ vkCmdResetQueryPoolArgs {
 
 sub void dovkCmdResetQueryPool(ref!vkCmdResetQueryPoolArgs args) {
   pool := QueryPools[args.QueryPool]
-  for i in (0 .. args.QueryCount) {
-    pool.Status[args.FirstQuery + i] = QUERY_STATUS_INACTIVE
+  if pool != null {
+    for i in (0 .. args.QueryCount) {
+      pool.Status[args.FirstQuery + i] = QUERY_STATUS_INACTIVE
+    }
+    pool.LastBoundQueue = LastBoundQueue
   }
-  pool.LastBoundQueue = LastBoundQueue
 }
 
 @threadSafety("app")
@@ -230,7 +236,9 @@ cmd void vkCmdResetQueryPool(
 
 sub void dovkCmdWriteTimestamp(ref!vkCmdWriteTimestampArgs args) {
   pool := QueryPools[args.QueryPool]
-  pool.LastBoundQueue = LastBoundQueue
+  if pool != null {
+    pool.LastBoundQueue = LastBoundQueue
+  }
 }
 
 @threadSafety("app")
@@ -266,7 +274,9 @@ class vkCmdCopyQueryPoolResultsArgs {
 
 sub void dovkCmdCopyQueryPoolResults(ref!vkCmdCopyQueryPoolResultsArgs args) {
   pool := QueryPools[args.QueryPool]
-  pool.LastBoundQueue = LastBoundQueue
+  if pool != null {
+    pool.LastBoundQueue = LastBoundQueue
+  }
 }
 
 @threadSafety("app")

